### ISSUE TITLE
 Constant in a Java interface that is then implemented in a concrete class causes a -Wprotocol error

### ISSUE THREAD

**Issue by confile** [google/j2objc#550]
Having set the ``--static-accessor-methods`` flag. When you implement a constant in a Java interface like this:  
```
public interface CameraManager {

  public enum SourceType {
    CAMERA_IMAGE, ALBUM_IMAGE
  }

  public static final boolean ALLOW_EDIT = true;
  public static final boolean SAVE_TO_PHOTO_ALBUM = true;

  void getPhotoWorkaround();
}
```
You get the following error while compiling the code outside of Xcode: 
```
error: method 'ALLOW_EDIT' in protocol 'BuddyisCoreAppPhotoEditorCameraManagerCameraManager' not implemented [-Werror,-Wprotocol]
@implementation BuddyisCoreAppPhotoEditorCameraManagerCameraManagerImpl
```

We discussed this issue while developing the j2objc-gradle plugin here: https://github.com/j2objc-contrib/j2objc-gradle/issues/151

**Comment by kstanger** [comment]
We're mistakenly putting the static accessor declaration within the @protocol declaration for interface types. It needs to move the the @interface declaration instead. (In this case the interface type will have both a ObjC protocol and a ObjC class)

I have a fix internally for this.

**Comment by lgaches** [comment]
When are you planning to fix this issue ?

**Comment by kstanger** [comment]
[Fixed since 0.9.8](https://github.com/google/j2objc/commit/d8172732456b8be61bc12b74322ecbe9a958f301).
